### ISSUE TITLE
[Alert Dialog]: issue with non-interactive page after closing

### ISSUE THREAD

**Issue by mattiafilardibelka** [radix-ui/primitives#2416]
### Current Behavior

When an Alert Dialog is displayed and then closed, it appends the `pointer-events: none` class on the body, making the entire page non-interactive.

![image](https://github.com/radix-ui/primitives/assets/116554671/d2f0e0c0-1896-45fa-bd11-a34caad66030)

After closing (both by pressing cancel and action buttons)

![image](https://github.com/radix-ui/primitives/assets/116554671/7beb752d-61fb-4732-95f5-188bea60d824)

### Expected behavior

The Alert Dialog should not affect the interactivity of the page after it is closed. The `pointer-events: none` class should be removed or not added in the first place.

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |    @radix-ui/react-alert-dialog     |     1.0.5    |
| React            | n/a     |     18.2.0    |
| Browser          |    chrome     |         |
| Assistive tech   |         |         |
| Node             | n/a     |     18.13.0    |
| npm/yarn         |    yarn     |    3.6.3     |
| Operating System |    macOS     |    13.2.1     |

**Comment by shayant98** [comment]
Also experiencing this issue!

Scenario:
Opening a Dialog from a button adds the following css property:
```css
pointer-events: none
```
when closing the toggle gets removed and the content below the dialog is interactable.

This is fine.

2nd scenario
Opening a dialog from a dropdown menu also adds the same css property to the main body but opening the dialog removes it.
After closing the dialog the pointer event gets added again.

It seems the functionality is to add/remove the CSS property is being toggled on every open/close event, but does not check what kind of event it is.

**Comment by edward1127** [comment]
I experience the same issue

**Comment by stimw** [comment]
same problem

**Comment by stimw** [comment]
> How are you composing Dialog + DropdownMenu?

hi, I just found your answer here: https://github.com/radix-ui/primitives/issues/837#issuecomment-1455160154

I tried several ways to compose dialog and dropdown-menu. It turns out that the `Dialog` must be wrapped in the `DropdownMenu.Content`. And finally it works.

It would be great to have the example in the docs cause this may be a common problem, and I spent a lot of time looking for relevant issue and solutions.

Thanks for you help.
